### PR TITLE
Fix Funding Source Exception, Item ID and Service Instance Id Generation

### PR DESCRIPTION
**Pull request by leerogers-bjss**
Fix funding source to not show partially completed items (to stop throwing an exception) - #20609

Fix funding source to calculate the total amount using the contract length

Fix Review pages to show ItemId as a global counting number

Fix review page service instance ID to always show and generate correctly - #20603

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Views/OrderSummary/OrderItem.cshtml [hunk 1]
```diff
@@ -51,29 +52,29 @@
...
-                </td>
-
-                @if (hasServiceRecipientQuantities)
-                {
-                    <td class="numeric">
-                        @(recipient.Quantity.HasValue ? $"{recipient.Quantity:N0}" : "-")
-                    </td>
-                }
-            </tr>
+			<tr>
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Views/OrderSummary/OrderItem.cshtml]: Lots of unnecessary formatting/spacing changes

**Review by bjss-colin — CHANGES_REQUESTED** [review]

**Review by magnusjohn-bjss — APPROVED** [review]

**Review by bjss-colin — APPROVED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Extensions/HttpContextExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+{
+    public static class HttpContextExtensions
+    {
+        public const string LineIdKey = "view-context-line-id";
+
+        public static int NextLineId(this HttpContext context)
+        {
+            var output = (int)(context.Items[LineIdKey] ?? 0);
+
+            context.Items[LineIdKey] = output + 1;
```
Comment (magnusjohn-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Extensions/HttpContextExtensions.cs:13]: This is going to mean the ids start at 0, is that correct?

Comment (leerogers-bjss) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Extensions/HttpContextExtensions.cs:13]: yes, thats what they do currently
